tests/lib/nested.sh

```diff
@@ -1231,6 +1221,14 @@ nested_exec_as() {
     sshpass -p "$PASSWD" ssh -p "$NESTED_SSH_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$USER"@localhost "$@"
 }
 
+nested_retry() {
+    if ! nested_exec "test -e ./retry" &>/dev/null; then
+        nested_copy "$TESTSTOOLS/retry"
+        nested_exec "sed 's/any-python/python3/g' -i ./retry"
```

Comment:
Can we just modify our retry code to use python3 and avoid this hack? is there a reason to have any-python?

If we stil have to do it, then would it be faster to make a local copy of retry, modify it with sed, then push a ready to use version to the nested vm, saving one nested_exec?

Reply:
I think some systems did not have python3 or it was super old (14.04?). Maybe we should revisit this and fix the tool to use python3 always.

Reply:
right, in that case it would be nice to remove all the any-python from the tools I think
I think I could that in a following or, does it make sense?

Reply:
Sounds good as a followup.

Reply:
I've opened #10523